cpp/find_warnings.py

```diff
@@ -433,8 +441,10 @@ def _find_unused_warnings(self, included_files, forward_declarations,
                 self._add_warning(msg, node)
 
     def _find_header_warnings(self):
-        included_files, forward_declarations = self._read_and_parse_includes()
-        self._find_unused_warnings(included_files, forward_declarations)
+        self._find_using_namespaces()
+        if(len(self.warnings) == 0):
```

Comment:
It seems like it would be better if `self._find_using_namespaces()` returned `True` if it finds a problem. That way you could do something like:

```python
if not self._find_using_namespaces():
    included_files, forward_declarations = self._read_and_parse_includes()
    self._find_unused_warnings(included_files, forward_declarations)
```

The above doesn't rely on the `self.warnings`, which might be modified by other methods (at least in future implementations).